pkg/daemon/writer.go

```diff
@@ -39,6 +39,7 @@ type clusterNodeWriter struct {
 type NodeWriter interface {
 	Run(stop <-chan struct{})
 	SetDone(client corev1client.NodeInterface, lister corev1lister.NodeLister, node string, dcAnnotation string) error
+	SetLayeredDone(client corev1client.NodeInterface, lister corev1lister.NodeLister, node string, image string) error
```

Comment:
When I hacked on this I was trying to decide if I should find a way to mangle SetDone to satisfy both paths, but I am very in favor of this being separate for now. 

We're eventually going to have to talk about the pool/node-controller consequences of how to get from where we are now with layering which is: 
- Our pool gets to a MachineConfig _via an image_

To the desired and state where: 

- Our pool gets to an Image (_that just so happens to contain the required machineconfig_)

Like eventually **the image needs to be the config**, not the rendered-config. 

Right now pool.Spec.Configuration.Name is implied to be a MachineConfig. As part of the build controller-y stuff I was thinking through what that might mean if the pool's config is an actual image. 

Do we just add a "Kind" there in the pool status/spec to go along with the name? And then the pool/node-controller is like "I got you to that config you specified, whatever kind it was". Definitely a longer discussion that crosses into UX.  

This comment is too long, but it's too late.

Reply:
Hmm yeah I'm less familiar with the controller side, I think I'd like to talk through some of that. From the MCD side, I think it would be nice if we could drop the SetDone call, because then all the MCD has to care about is 1. What's my DesiredImageConfigAnnotationKey and 2. did I set MachineConfigDaemonStateAnnotationKey and CurrentImageConfigAnnotationKey